bin/ch/ch.cpp

```diff
@@ -196,48 +207,536 @@ static void CALLBACK PromiseContinuationCallback(JsValueRef task, void *callback
-    // TODO: Remove this code in a future iteration once Utf8 versions of the Jsrt API is implemented
-    IfFailGo(Helpers::NarrowStringToWideDynamic(fullPath, &fullPathWide));
+    size_t fplength = (wcslen(path) + 2);
+    char16* fullpath = new char16[fplength];
+    fullpath[0] = _u('\0');
 
-    JsErrorCode runScript;
-    if (bcBuffer != nullptr)
+    wcscat_s(fullpath, fplength, path);
+    if(!isPathDirName)
     {
-        runScript = ChakraRTInterface::JsRunSerializedScript(fileContents, bcBuffer, WScriptJsrt::GetNextSourceContext(), fullPathWide, nullptr /*result*/);
+        wcscat_s(fullpath, fplength, _u("\\"));
+    }
+
+    DWORD dwAttrib = GetFileAttributes(fullpath);
+    if((dwAttrib != INVALID_FILE_ATTRIBUTES) && (dwAttrib & FILE_ATTRIBUTE_DIRECTORY))
+    {
+        delete[] fullpath;
+        return;
+    }
+
+    BOOL success = CreateDirectory(fullpath, NULL);
+    if(!success)
+    {
+        DWORD lastError = GetLastError();
+        LPTSTR pTemp = NULL;
+        FormatMessage(FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_ARGUMENT_ARRAY, NULL, lastError, 0, (LPTSTR)&pTemp, 0, NULL);
+        fwprintf(stderr, _u(": %s"), pTemp);
+
+        AssertMsg(false, "Failed Directory Create");
+    }
+
+    delete[] fullpath;
+#endif
+}
+
+void DeleteDirectory(const char16* path)
+{
+#ifndef _WIN32
+    AssertMsg(false, "Not XPLAT yet.");
+#else
+    HANDLE hFile;
+    WIN32_FIND_DATA FileInformation;
+
+    bool isPathDirName = (path[wcslen(path) - 1] == _u('\\'));
+
+    size_t splength = (wcslen(path) + 5);
+    char16* strPattern = new char16[splength];
+    strPattern[0] = _u('\0');
+
+    wcscat_s(strPattern, splength, path);
+    if(!isPathDirName)
+    {
+        wcscat_s(strPattern, splength, _u("\\"));
+    }
+    wcscat_s(strPattern, splength, _u("*.*"));
+
+    hFile = ::FindFirstFile(strPattern, &FileInformation);
+    if(hFile != INVALID_HANDLE_VALUE)
+    {
+        do
+        {
+            if(FileInformation.cFileName[0] != '.')
+            {
+                size_t sfplength = (wcslen(path) + wcslen(FileInformation.cFileName) + 2);
+                char16* strFilePath = new char16[sfplength];
+                strFilePath[0] = _u('\0');
+
+                wcscat_s(strFilePath, sfplength, path);
+                if(!isPathDirName)
+                {
+                    wcscat_s(strFilePath, sfplength, _u("\\"));
+                }
+                wcscat_s(strFilePath, sfplength, FileInformation.cFileName);
+
+                if(FileInformation.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY)
+                {
+                    DeleteDirectory(strFilePath);
+                    ::RemoveDirectory(strFilePath);
+                }
+                else
+                {
+                    // Set file attributes
+                    ::SetFileAttributes(strFilePath, FILE_ATTRIBUTE_NORMAL);
+                    ::DeleteFile(strFilePath);
```

Comment:
Nothing bad. We will overwrite important files later but some useless files may remain. This is mainly to keep things clean.